src/pages/Login.jsx

```diff
@@ -40,26 +48,24 @@ class LoginPage extends Component {
                         type="text" 
                         name="username"
                         value={username} 
-                        onInput={/* setState */}
+                        onInput={e => this.setState({ username: e.target.value }) }
                         />
                 </div>
                 <div className={'login-field' + 
                 (submitted && !password ? 'has-error' : '')}>
                     <label htmlFor="password">Password</label>
                     <input 
-                        type="text" 
+                        type="password" 
                         name="password"
                         value={password} 
-                        onInput={/* setState */}
+                        onInput={e => this.setState({ password: e.target.value }) }
                         />
                 </div>
                 <button
                     className={`submit-button${!(password && username) ? ' disabled' : ''}`}
                     type="button"
                     disabled={!password || !username}
-                    onClick={() => {
-                        //Trigger submit event for reducer
-                    }}
+                    onClick={this.handleSubmit}
```

Comment:
@GerryFudd I didn't totally understand your recommendation here, so I just added a simple event handler to deal with it until we replace it with something else.